### PR TITLE
Fixes #21243 - replace alias_method_chain

### PR DESCRIPTION
**Pull request by jlsherrill**
This replaces alias_method_chain instances
with use of prepend, and sets a convention
for overriding  of foreman methods

### CODE REVIEW

**Comment by jlsherrill** [comment]
[test]

**Comment by mmoll** [comment]
this needs a rebase...

**Comment by jlsherrill** [comment]
and a review :)

File: test/support/pulp/task_support.rb [hunk 1]
```diff
@@ -23,21 +23,5 @@ def self.wait_on_tasks(task_list, options = {})
...
-      end
-
-      module ClassMethods
-        def any_task_running_with_vcr(async_tasks)
-          VCR.live? ? any_task_running_without_vcr(async_tasks) : false
-        end
-      end
-    end
-
-    Katello::PulpTaskStatus.send(:include, TaskRunningWithVcr)
```
Comment (akofink) [inline test/support/pulp/task_support.rb]: I get what's happening in this PR, but this change breaks the pattern. I guess I'm not sure why we need to remove this. Could you please explain?

**Comment by johnpmitsch** [comment]
I'm not sure what is best practices, but are there concerns with putting the Override module in a separate file? (Other than method names  overlapping)

Otherwise looks fine

**Comment by jlsherrill** [comment]
@johnpmitsch i contemplated that, but then we'd have all these 2nd files just to hold one little method.  It made more sense to me to put them together, especially since overrides is likely to be small.

**Pull request merged**